File: Programmers/Lv0/전국대회선발고사.py
def solution(rank, attendance):
    dic = {}
    i = 0
    for r in rank:
        if attendance[i] == True:
            dic[r] = i
        i += 1
    tmp = sorted(list(dic.keys()))
    return 10000 * dic[tmp[0]] + 100 * dic[tmp[1]] + dic[tmp[2]]

File: Programmers/Lv0/전국대회선발고사.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> rank, vector<bool> attendance) {
    int _map[rank.size() + 1];
    for (int i = 0; i < rank.size() + 1; i++)
        _map[i] = -1;

    int index = 0;
    for (int r : rank) {
        if (attendance[index] == true)
            _map[r] = index;
        ++index;
    }

    int ret[3];
    int j = 0;
    for (int idx = 0; idx < rank.size() + 1 && j != 3; idx++) {
        if (_map[idx] != -1) {
            ret[j] = _map[idx];
            j++;
        }
    }
    return 10000 * ret[0] + 100 * ret[1] + ret[2];
}